test/test_transforms_v2_refactored.py

```diff
@@ -125,6 +128,7 @@ def check_kernel(
     check_cuda_vs_cpu=True,
     check_scripted_vs_eager=True,
     check_batched_vs_unbatched=True,
+    expect_same_dtype=True,
```

Comment:
Not against it, but are we expecting more kernels to set this to False?

Reply:
I don't think so (more precisely: I don't know). We still want the rest of the checks to be done for ToDtype though. Is there a better way than to add a parameter?

Reply:
I have a slight preference for adding the parameter, because otherwise we'd have to change both implementation of `check_kernel` if we ever needed to update it.
(looks like I'm advocating for this single entry point after all :p )